File: src/main/java/net/nodus/security/auth/service/client/GoogleTokenClient.java
package net.nodus.security.auth.service.client;

import java.util.Map;
import net.nodus.security.auth.service.GoogleOAuthService.GoogleTokenInfo;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.PostMapping;

@FeignClient(name = "googleTokenClient", url = "https://oauth2.googleapis.com")
public interface GoogleTokenClient {

    @PostMapping(value = "/token", consumes = MediaType.APPLICATION_FORM_URLENCODED_VALUE)
    GoogleTokenInfo exchangeCode(Map<String, String> params);

}

File: src/main/kotlin/net/nodus/auth/service/client/GoogleTokenClient.kt
package net.nodus.auth.service.client

import org.springframework.cloud.openfeign.FeignClient
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestHeader

@FeignClient(name = "googleTokenClient", url = "https://oauth2.googleapis.com")
interface GoogleTokenClient {
    @PostMapping(value = ["/token"], consumes = [MediaType.APPLICATION_FORM_URLENCODED_VALUE])
    fun exchangeCode(params: Map<String, *>): Map<String, Any>
}

@FeignClient(name = "googleUserInfoClient", url = "https://openidconnect.googleapis.com")
interface GoogleUserInfoClient {
    @GetMapping("/v1/userinfo")
    fun fetchUserInfo(@RequestHeader("Authorization") bearerToken: String): Map<String, Any>
}